DigitalLearningSolutions.Web/Views/MyAccount/EditDetails.cshtml

```diff
@@ -1,86 +1,86 @@
-                                  will not be updated."; }
-      <input type="hidden" asp-for="ProfileImage"/>
-      <vc:file-input asp-for="ProfileImageFile" label="Profile picture (optional)" hint-text="@hintText" css-class="nhsuk-u-width-full"/>
-    </div>
-    <div class="nhsuk-grid-column-one-half">
-      <div class="profile-picture__image-box">
-        @if (Model.ProfileImage != null) {
-          <img class="profile-picture__image" src="data:image;base64,@Convert.ToBase64String(Model.ProfileImage)" alt="Profile Picture"/>
-        } else {
-          <img class="profile-picture__image" src="@Url.Content("~/images/avatar.png")" alt="Default Profile Picture"/>
-        }
-      </div>
-    </div>
-  </div>
-
-  <div class="nhsuk-grid-row divider">
-    <div class="nhsuk-grid-column-one-half">
-      <button name="action" class="nhsuk-button" value="previewImage">Preview</button>
-      <button name="action" class="nhsuk-button nhsuk-button--secondary" value="removeImage">Remove</button>
-    </div>
-  </div>
-
-  <div class="nhsuk-grid-row">
-    <div class="nhsuk-grid-column-full">
-      <vc:text-input asp-for="Password" label="Confirm password to save changes:" type="password" spell-check="false" autocomplete="" hint-text="" css-class="nhsuk-u-width-one-half"/>
-
-      <button name="action" class="nhsuk-button" value="save">Save</button>
-    </div>
-  </div>
-</form>
-<div class="nhsuk-grid-row">
-  <div class="nhsuk-grid-column-full">
-    <vc:back-link asp-controller="MyAccount" asp-action="Index"/>
-  </div>
-</div>
+﻿@using DigitalLearningSolutions.Web.ViewModels.MyAccount
+@model EditDetailsViewModel
+
+<link rel="stylesheet" href="@Url.Content("~/css/myAccount/myAccount.css")" asp-append-version="true">
+
+@{
+  var errorHasOccurred = !ViewData.ModelState.IsValid;
+  ViewData["Title"] = errorHasOccurred ? "Error: Edit Details" : "Edit Details";
+}
+
+<div class="nhsuk-grid-row">
+  <div class="nhsuk-grid-column-full">
+    <h1 class="nhsuk-heading-xl" id="app-page-heading">Edit details</h1>
+  </div>
+</div>
+
+<form class="nhsuk-u-margin-bottom-8" method="post" novalidate asp-action="EditDetails" enctype="multipart/form-data">
+  <div class="nhsuk-grid-row">
+    <div class="nhsuk-grid-column-full">
+      @if (errorHasOccurred) {
+        <vc:error-summary order-of-property-names="@(new[] { nameof(Model.FirstName), nameof(Model.LastName), nameof(Model.Email), nameof(Model.Password) })"/>
+      }
+    </div>
+  </div>
+
+  <div class="hidden-submit">
+    <button name="action" class="nhsuk-button" value="save">Save</button>
+  </div>
+
+  <div class="nhsuk-grid-row">
+    <div class="nhsuk-grid-column-full">
+      <vc:text-input asp-for="FirstName" label="First name" value="@Model.FirstName" type="text" spell-check="false" autocomplete="given-name" hint-text="" css-class="nhsuk-u-width-one-half"/>

```

Comment:
Not quite sure why the value attribute has been added here. Is it something to do with the merge conflicts?

Reply:
Looking back at the commits, I think it is to stop values that shouldn't be getting returned after posting, e.g. the password field. 

I suspect the merge conflict currently is just the line ending, I have no idea why we keep getting these, but it must be either myself or Steve at this point. I have git configured as Steve said in slack (git config core.autocrlf = false), but since these are all files that I created, something is clearly going wrong.

That being said, there may be future merge conflicts that one of us will have to deal with. I am in the process of adding another field to this page that uses the same text input component.

Reply:
@AlexJacksonDS Spot on! It's because the value was being set explicitly in each text-input. The extra field allows you to choose whether the value gets set.
By the way Alex, is your IDE set to use CRLF line endings as well?

Reply:
I believe so, new files I created in the last couple of days seem to be CRLF. I have no idea why some that I have previously created aren't. I'm using VS, which I think should be using windows default

Reply:
But I've decided it would improve it to use a flag instead and reduce data handling duplication.